packages/axios-extension/src/abap/abap-service-provider.ts

```diff
@@ -153,4 +170,52 @@ export class AbapServiceProvider extends ServiceProvider implements AbapServiceP
         }
         return this.services[LayeredRepositoryService.PATH] as LayeredRepositoryService;
     }
+
+    /**
+     *
+     * @param packageName Package name for deployment
+     * @param appName Fiori project name for deployment. A new project that has not been deployed before is also allowed
+     * @param schema Decorated para that will be autofilled
+     * @returns
+     */
+    @adt(AdtServiceConfigs[AdtServiceName.TransportChecks])
```

Comment:
I don't understand the `@adt` decorator abstraction. Can't we use an `ADTServiceCatalog` or something that takes care of calling discovery and loading the schema as required? This feels too implicit/magic to me. Feels like the name `@adt` could be more descriptive too. Something along the lines of `@usesAdtService()`. Even then I don't see the point of exposing the concept of the ADT Schema to endusers. From what I can see users will try and access ADT services, discovery and schema loading are incidental to this purpose. It's also easy to miss using the decorator. If a catalog object were used instead, users would not miss dealing with an instance of it